libexec/rtld-elf/cheri/cheri_reloc.h

```diff
@@ -75,7 +78,7 @@ process_r_cheri_capability(Obj_Entry *obj, Elf_Word r_symndx,
 	assert(ELF_ST_TYPE(def->st_info) != STT_GNU_IFUNC &&
 	    "IFUNC not implemented!");
 
-	const void * __capability symval = NULL;
+	void * __capability symval = NULL;
```

Comment:
Do the propagation, or `__DECONST` if you really have to. I don't like changing code outside c18n just because your code can't be bothered to handle what's there today properly.